### PR TITLE
Allow organization policies to be specified as YAML files

### PR DESCRIPTION
**Pull request by srivasta**
This addresses Issue #627

Signed-off-by: Manoj Srivastava <srivasta@google.com>

### CODE REVIEW

**Comment by rosbo** [comment]
This issue talks about adding support for YAML file but the code actually adds support for JSON file

**Comment by srivasta** [comment]
Aargh. I do read problem descriptions most of the time, I swear. Change
upcoming. Will also look into what tests failed

  Manoj

File: google/resource_google_organization_policy.go [hunk 1]
```diff
@@ -99,6 +114,56 @@ var schemaOrganizationPolicy = map[string]*schema.Schema{
 		Type:     schema.TypeString,
 		Computed: true,
 	},
+	// Detect changes to local file or changes made outside of Terraform to the file stored on the server.
+	"detect_policy_change": &schema.Schema{
```
Comment (ndmckinley) [inline google/resource_google_organization_policy.go]: I'm relatively new to Terraform, but I haven't seen this pattern before.  Is this something that you've seen done elsewhere?

File: google/resource_google_organization_policy.go [hunk 2]
```diff
@@ -194,16 +259,35 @@ func resourceGoogleOrganizationPolicyImportState(d *schema.ResourceData, meta in
...
+		}
+	} else {
+		listPolicy, err = expandListOrganizationPolicy(d.Get("list_policy").([]interface{}))
+	}
 	if err != nil {
 		return err
 	}
 
+	if source, ok := d.GetOkExists("boolean_policy_source"); ok {
+		filePolicy, err := getFilePolicy(source.(string))
```
Comment (ndmckinley) [inline google/resource_google_organization_policy.go]: You need to return this `err`, right?

File: website/docs/r/google_organization_policy.html.markdown [hunk 3]
```diff
@@ -62,6 +71,18 @@ resource "google_organization_policy" "services_policy" {
...
 
+You can also provide the policy in YAML format:
+
+
+```hcl
+resource "google_folder_organization_policy" "services_policy" {
+  org_id     = "123456789"
+  constraint = "serviceuser.services"
+
+  list_policy_file = "path/to/policy_file.json"
```
Comment (ndmckinley) [inline website/docs/r/google_organization_policy.html.markdown]: Should probably say `.yml`

File: website/docs/r/google_organization_policy.html.markdown [hunk 4]
```diff
@@ -28,6 +28,16 @@ resource "google_organization_policy" "serial_port_policy" {
...
 ```
 
+Alternately, you can write the policy in YAML format and provide the source file path:
+
+```hcl
+resource "google_folder_organization_policy" "serial_port_policy" {
+  org_id     = "123456789"
+  constraint = "compute.disableSerialPortAccess"
+
+  boolean_policy_source = "path/to/file.json"
```
Comment (ndmckinley) [inline website/docs/r/google_organization_policy.html.markdown]: `.yml`?

File: google/test-fixtures/google_organization_policy_bool.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (ndmckinley) [inline google/test-fixtures/google_organization_policy_bool.json]: Better to test with .yml files, even though yml is a superset of json.

**Comment by sethvargo** [comment]
I'm 👎 on this. There are many tools that will convert that YAML to JSON (or another language), and I don't think Terraform should take on that burden.

**Comment by ndmckinley** [comment]
Agreed.  Without further responses, closing - please feel free to come back to this at a later date and open a new issue, if you would like to discuss Seth's objection.

**Pull request closed without merging**